README.md

```diff
@@ -81,14 +81,41 @@ constructs the case class specified in its type constraint. To do this,
 building inner validated case class instances or from using the
 `validate` or `unchecked` methods to validate values at a given path.
 
-### Copy-free Validated `Config` Instances
+### Secure Validated `Config` Instances
 
 Sometimes, after validating and building a `Config` instance, we do not
 wish the constructed case class instance to be modified (e.g. by calling
 a `copy` constructor) - allowing this could allow validation guarantees
 to be broken! To support such scenarios, one simply extends the trait
 `CopyFreeConfig` when defining the case class.
 
+Secure configuration objects may then be constructed as follows:
+```scala
+object SecureConfigBuilder {
+  abstract case class SecureHttpConfig private[SecureConfigBuilder] (host: String, port: Int) extends CopyFreeConfig[SecureSettings]
```

Comment:
It may be sufficient to simply supply the abstract modifier here (i.e. can we get rid of a need for `CopyFreeConfig`)?